### DISCUSSION TITLE
Expand/Collapse not working and always getting error "resizeobserver loop limit exceeded"

### DISCUSSION THREAD

**Discussion by optic14** [xyflow/xyflow#3018] [Q&A]
i have made a mindmap using reactflow and dagre but now i am stuck at inserting expand/collapse feature in it. I am always getting "resizeobserver loop limit exceeded".
following is my handleNodeExpandCollapse() function which basically fetches node id and then if the node has isExpanded property true, it sets it to false for the subnodes of that nodeid, and then re-renders the layout with the new nodes i.e. collapsed nodes.

    const handleNodeExpandCollapse = (nodeId) => {
    const updatedNodes = nodes.map((node) => {
      if (node.id === nodeId) {
        return {
          ...node,
          data: {
            ...node.data,
            isExpanded: !node.data.isExpanded,
          },
        };
      }
      return node;
    });
    const { nodes: newNodes, edges: newEdges } = createNodesAndEdges(updatedNodes);
    const layoutedElements = getLayoutedElements(newNodes, newEdges, layoutDirection);
    setNodes([...layoutedElements.nodes]);
    setEdges([...layoutedElements.edges]);
  };

//this is my nodetypes for customNode props...

  const nodeTypes = useMemo(() => ({
    custom: (props) => (
      <CustomNode
        {...props}
        layoutDirection={layoutDirection}
        subnodes={props.data.subnodes}
        onExpandCollapse={handleNodeExpandCollapse}
      />
    ),
  }), [layoutDirection]);

**Comment by moklick** [comment]
Please provide a codesandbox https://new.reactflow.dev/react-js
